.github/workflows/generic_build.yml

```diff
@@ -0,0 +1,228 @@
+        repository: 'bioconda/bioconda-utils'
+        path: 'images/bioconda-utils-build-env-cos7/bioconda-utils'
+
+    - name: Build
+      id: build
+      run: |
+        set -xeu
+
+        # Store provided inputs as local vars
+        image_name='${{ inputs.image_prefix }}${{ inputs.image_name }}'
+        image_dir='images/${{ inputs.image_name }}'
+        tags='${{ inputs.tags }}'
+        debian_version='${{ inputs.debian_version }}'
+        busybox_version='${{ inputs.busybox_version }}'
+        archs='${{ inputs.archs }}'
+        bioconda_utils='${{ inputs.bioconda_utils }}'
+
+        # Adds image and tags to outputs which can be used in later steps.
+        printf %s\\n \
+          "image_dir=${image_dir}" \
+          "image_name=${image_name}" \
+          "tags=$( echo ${tags} )" \
+          >> $GITHUB_OUTPUT
+
+        # Dockerfile lives here
+        cd $image_dir
+
+        for tag in ${tags} ; do
+          buildah manifest create "${image_name}:${tag}"
+        done
+
+        # get space-separated archs input string into an array
+        read -r -a archs_and_images <<<"$archs"
+
+        # ----------------------------------------------------------------------
+        # Incrementally compose build args, depending on which inputs were
+        # provided.
+        BUILD_ARGS=()
+        if [ ! -z "${debian_version}" ]; then
+          BUILD_ARGS+=("--build-arg=debian_version=$debian_version")
+        fi
+
+        if [ "${bioconda_utils}" == "true" ]; then
+        # Due to different nomenclature used by conda-forge and buildah, we
+        # need to map archs to base images.
+          archs_and_images=(
+            "amd64=quay.io/condaforge/linux-anvil-cos7-x86_64"
+            "arm64=quay.io/condaforge/linux-anvil-aarch64"
+          )
+        fi
+
+        # If busybox_version was specified, assume we are making a busybox image.
+        if [ ! -z "${busybox_version}" ]; then
+          BUILD_ARGS+=("--build-arg=busybox_version=$busybox_version")
+
+          # Make a busybox image that we'll use further below. As shown in the
+          # Dockerfile.busybox, this uses the build-busybox script which in turn
+          # cross-compiles for x86_64 and aarch64, and these are later copied
+          # into an arch-specific container.
+          # --iidfile prints the built image ID to the specified file so we can
+          # refer to the image later
+          iidfile="$( mktemp )"
+          buildah bud \
+            --iidfile="${iidfile}" \
+            --file=Dockerfile.busybox \
+            ${BUILD_ARGS[@]}
+          busybox_image="$( cat "${iidfile}" )"
+          rm "${iidfile}"
+
+          # And then extend the build args with this image.
+          BUILD_ARGS+=("--build-arg=busybox_image=${busybox_image}")
+        fi
+
+        # ----------------------------------------------------------------------
+
+        # Build each arch's image
+        for arch in $archs_and_images ; do
```

Comment:
Currently only an AMD64 image is added to the manifest - https://pipelinesghubeus24.actions.githubusercontent.com/0YHXh0QGzpCiRgaqrcqmDx4aoYjcAOBDwhzeyyC598TlkSuHqS/_apis/pipelines/1/runs/345/signedlogcontent/2?urlExpires=2024-02-05T15%3A40%3A20.3157368Z&urlSigningMethod=HMACV1&urlSignature=hK6kHUewgqlYl8%2BGIobY20GzaihVEyRhvkhn2ZBPr00%3D.
I think this should be `for arch in "${archs_and_images[@]}"; do` to iterate over the strings in the Bash array.